cockpit/gui/mosaic/window.py

```diff
@@ -87,14 +87,32 @@
 SIDEBAR_WIDTH = 150
 
 ## Timeout for mosaic new image events
-CAMERA_TIMEOUT = 5
+CAMERA_TIMEOUT = 1
 ##how good a circle to draw
 CIRCLE_SEGMENTS = 32
 PI = 3.141592654
 
 ## Simple structure for marking potential beads.
 BeadSite = collections.namedtuple('BeadSite', ['pos', 'size', 'intensity'])
 
+from functools import wraps
+
+
+def _pauseMosaicLoop(func):
+    @wraps(func)
+    def wrapped(self, *args, **kwargs):
+        is_running = self.shouldContinue.is_set()
+        if not is_running:
+            return func(self, *args, **kwargs)
+        else:
+            self.shouldContinue.clear()
+            try:
+                return func(self, *args, **kwargs)
+            except:
+                raise
+            finally:
+                self.shouldContinue.set()
```

Comment:
I don't understand the purpose of this `except`. The block in `finally` is handled after the handling in `except` but that just re-raises it. I would think the `except` block can be removed but maybe I'm just missing something?